README.md

```diff
@@ -63,1498 +104,468 @@ Delegating a capability is like the statement `message`. Task is akin to `messag
-    }
-  ]
-}
-```
+1. Specifying a particular `sub` (and thus `aud`) when [enqueuing new Tasks][enqueue] in a Receipt
+2. Indexing Receipts for reverse lookup and memoization
 
-### 7.3.2 Effect continuing thread execution
+#### Audience
+[Audience]: #audience
 
-```json
-{
-  "join": {
-    "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
-  }
-}
-```
+The OPTIONAL `aud` field specified the intended recipient of Invocation, otherwise the Audience MUST be assumed to the [Subject]. This is useful for message routing, command brokers, proxy execution, gateways, replicated state machines, and so on.
 
-### 7.3.1 Effect with fork & join
+### Task
+[Task]: #task
 
-```json
-{
-  "join": {
-    "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
-  },
-  "fork": [
-    {
-      "/": "bafyreigmmdzix2vxboojvv6j6h7sgvxnrecdxtglwtqpxw7hybebzlsax4"
-    },
-    {
-      "/": "bafyreif6gfpzgxnii4ys6a4bjenefg737fb5bgam3onrbmhnoa4llk244q"
-    }
-  ]
-}
-```
-
-# 8 Receipt
-
-A `Receipt` is an attestation of the [Result] and requested [Effect]s by a [Task] [Invocation]. A Receipt MUST be signed by the [Executor] or it's delegate. If signed by the delegate, the proof of delegation from the [Executor] to the delegate (the `iss` of the receipt) MUST be provided in `prf`.
+A Task is the subset of Invocation fields that uniquely determine the work to be performed. The nonce is important for distinguishing between non-idempotent executions of a Task by making the group together unique.
+ 
+A Task MUST be uniquely defined by a Task ID that is the CID of the following fields:
 
-**NB: a Receipt does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
+- [Subject]
+- [Command]
+- [Arguments]
+- [Nonce]
 
-Receipts MUST use the same version as the invocation that they contain.
+Tasks that describe pure functions — or other strategies like fan-out racing — SHOULD have the same Task ID by using the same nonce.
 
-## 8.1 Schema
+#### Command
+[Command]: #command
 
-```ipldsch
-type Receipt struct {
-  ran     &Invocation
+The REQUIRED Command (`cmd`) field MUST contain a concrete, dispatchable message that can be sent to the Executor. The Command MUST define the shape of the data in the [Arguments].
 
-  # output of the invocation
-  out     Result
+### Arguments
+[Arguments]: #arguments
 
-  # Effects to be performed
-  fx      Effects
+The REQUIRED Arguments (`args`) field, MAY contain any parameters expected by the Command. The Subject MUST be considered the authority on the shape of this data. This field MUST be representable as a map or keyword list.
 
-  # All the other metadata
-  meta    {String: any}
+The Arguments MUST pass validation of the Policies on all of the [UCAN Delegations][UCAN Delegation] in the [Proofs] field. If any Policy reports failure, the Invocation MUST be rejected.
```

Comment:
Note that [Arguments] is not defined in the high-level spec either, and there are still references to "Policy" there worth CTRL-Fing to clean up once Arguments is added there (as optional, I assume?)

Reply:
Arguments is not a common field across all UCANs. It's special for Invocation.